libnavigation-core/src/main/java/com/mapbox/navigation/core/stops/ArrivalProgressObserver.kt

```diff
@@ -11,24 +11,39 @@ internal class ArrivalProgressObserver(
 ) : RouteProgressObserver {
 
     private var arrivalController: ArrivalController = AutoArrivalController()
+    private val arrivalObservers: MutableSet<ArrivalObserver> = mutableSetOf()
     private var arrivedForRoute = false
```

Comment:
Should this be atomic 🤔?

Reply:
hope not. the only caller is `onRouteProgressChanged`